### PR TITLE
Feike/use compiled artifacts

### PR DESCRIPTION
**Pull request by feikesteenbergen**
Before this commit, we compiled toolkit and promscale, multiple versions for multiple PostgreSQL versions.
The issue with that is that it was *very* time consuming, which isn't great for developing against this repository.

This was the situation before this commit:

```
         stage         | duration | pct  |  total   
-----------------------+----------+------+----------
 prep (apt-install)    | 00:04:06 |  1.6 | 04:10:33
 Cleanup               | 00:04:07 |  1.6 | 04:10:33
 Compiling Cloudutils  | 00:09:14 |  3.7 | 04:10:33
 Compiling TimescaleDB | 00:27:06 | 10.8 | 04:10:33
 Compiling Promscale   | 01:25:15 | 34.0 | 04:10:33
 Compiling Toolkit     | 02:00:46 | 48.2 | 04:10:33
 ```
 
 By taking out Toolkit and Promscale from the compilation, we expect a reduction of `34.0%+48.2% = 82.2%` in build times.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -231,7 +231,7 @@ RUN --mount=type=secret,uid=1000,id=private_repo_token \
 # - files owned by postgres can be overwritten in a running container
 # - new files can be added to the directories mentioned here
 RUN for pg in ${PG_VERSIONS}; do \
-        for dir in "$(/usr/lib/postgresql/${pg}/bin/pg_config --sharedir)/extension" "$(/usr/lib/postgresql/${pg}/bin/pg_config --pkglibdir)" "$(/usr/lib/postgresql/${pg}/bin/pg_config --bindir)"; do \
+        for dir in /usr/share/doc "$(/usr/lib/postgresql/${pg}/bin/pg_config --sharedir)/extension" "$(/usr/lib/postgresql/${pg}/bin/pg_config --pkglibdir)" "$(/usr/lib/postgresql/${pg}/bin/pg_config --bindir)"; do \
```
Comment (alexeyklyukin) [inline Dockerfile:234]: so you want manpages inside the container?

Comment (alexeyklyukin) [reply Dockerfile:234]: oh, possibly, we need /usr/share/doc to install dpkg packages?

Comment (feikesteenbergen) [reply Dockerfile:234]: No, they are actually removed afterwards, in the cleanup phase.

However, these `*.deb` files that we install *include* manpages, so we have to allow them to exist.

**Review by alexeyklyukin — APPROVED** [review]
Looks great!

File: build_scripts/install_promscale.sh [hunk 2]
```diff
@@ -20,25 +20,8 @@ export PATH="/usr/lib/postgresql/${PGVERSION}/bin:${PATH}"
...
-    if [ "${MAJOR}" -le 0 ] && [ "${MINOR}" -le 3 ]; then
-        PG_VER=pg${PGVERSION} make install || exit 1;
-    elif [ "${PROMSCALE_VERSION}" = "0.5.1" ]; then
-        make package && cp -v --recursive "./target/release/promscale-pg${PGVERSION}/"* / || exit 1
-    else
-        (make package && make install) || exit 1;
-    fi
+    DEBVERSION="$(apt info -a promscale-extension-postgresql-${PGVERSION} | awk '/Version:/ {print $2}' | grep "${PROMSCALE_VERSION}" | head -n 1)"
+    apt-get download promscale-extension-postgresql-${PGVERSION}=${DEBVERSION}
+    dpkg --install --admindir /tmp/dpkg --force-depends --force-not-root --force-overwrite promscale-extension-postgresql-${PGVERSION}*${DEBVERSION}*.deb
```
Comment (JamesGuthrie) [inline build_scripts/install_promscale.sh:25]: Unfortunately this installation method doesn't quite do what we want. We want to have all of the versioned shared objects on disk, but installing like this doesn't do that... It's possible/probable that the fault lies with how we've packaged our extension (it seems as though toolkit has the same problem).

The issue is that

Comment (feikesteenbergen) [reply build_scripts/install_promscale.sh:25]: Yes, I see only 1 version specific library exists. Going to try a few more things here though.

Comment (feikesteenbergen) [reply build_scripts/install_promscale.sh:25]: I removed the the dpkg directory in between installations; that does seem to give us what we want:

- for every minor version, a separate shared object.

*toolkit*
```bash
postgres@8161a70cf214:~$ find /usr/lib -name '*toolkit*.so'
/usr/lib/postgresql/14/lib/timescaledb_toolkit-1.6.0.so
/usr/lib/postgresql/14/lib/timescaledb_toolkit-1.8.0.so
/usr/lib/postgresql/14/lib/timescaledb_toolkit-1.7.0.so
/usr/lib/postgresql/12/lib/timescaledb_toolkit-1.6.0.so
/usr/lib/postgresql/12/lib/timescaledb_toolkit-1.8.0.so
/usr/lib/postgresql/12/lib/timescaledb_toolkit-1.7.0.so
/usr/lib/postgresql/13/lib/timescaledb_toolkit-1.6.0.so
/usr/lib/postgresql/13/lib/timescaledb_toolkit-1.8.0.so
/usr/lib/postgresql/13/lib/timescaledb_toolkit-1.7.0.so
```
*promscale*
```bash
postgres@8161a70cf214:~$ find /usr/lib -name '*promscale*.so'
/usr/lib/postgresql/14/lib/promscale-0.5.2.so
/usr/lib/postgresql/14/lib/promscale-0.5.1.so
/usr/lib/postgresql/14/lib/promscale-0.5.0.so
/usr/lib/postgresql/12/lib/promscale-0.5.2.so
/usr/lib/postgresql/12/lib/promscale-0.5.1.so
/usr/lib/postgresql/12/lib/promscale-0.5.0.so
/usr/lib/postgresql/13/lib/promscale-0.5.2.so
/usr/lib/postgresql/13/lib/promscale-0.5.1.so
/usr/lib/postgresql/13/lib/promscale-0.5.0.so
```

Comment (JamesGuthrie) [reply build_scripts/install_promscale.sh:25]: 🤯

**Review by JamesGuthrie — APPROVED** [review]

**Comment by jerryxwu** [comment]
@feikesteenbergen great to see this changes! 

Just wanted to confirm with you on one thing. As you know, the toolkit only started to build one DEB package for all starting from 1.6.0. Before this release, we had a version built specifically for timescale cloud. Are 1.6.0, 1.7.0 and 1.8.0 sufficient releases for you for now?

**Review by rtwalker — APPROVED** [review]

**Comment by feikesteenbergen** [comment]
> Are 1.6.0, 1.7.0 and 1.8.0 sufficient releases for you for now?

Yes, for us, they are.

**Pull request merged**